tests/test_article_parser.py

```diff
@@ -184,4 +194,49 @@ def test__ignore_code_blocks(self, root):
         assert article.lines[9].links[0].raw_location == "/wiki/Gameplay/Break"
 
         assert len(article.lines[11].links) == 1
-        assert article.lines[11].links[0].raw_location == "/wiki/Fun_stuff"
\ No newline at end of file
+        assert article.lines[11].links[0].raw_location == "/wiki/Fun_stuff"
+
+
+class TestFrontMatter:
+    def test__read_write(self, root):
+        article_path = root.join("en.md")
+        article_path.write_text(textwrap.dedent('''
+            ---
+            outdated: true
+            tags:
+              - a
+              - aaa
+              - юниcode
+            ---
+
+            # Test
+
+            Lorem (ipsum).
+        '''), encoding='utf-8')
+
+        with article_path.open("r") as fd:
```

Comment:
you always gotta specify the encoding to account for windows (this causes the test to fail)

```suggestion
        with article_path.open("r", encoding='utf-8') as fd:
```